### PR TITLE
test: Improve errors for missing/unused HTTP mocks

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Ease the frustrating experience of writing HTTP mocks by enhancing the error messages.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Before:

image

After:

image

**Comment by zharinov** [comment]
Before:

image

After:

image

**Comment by zharinov** [comment]
# Before

Code 2024-09-29 19 51 47

Code 2024-09-29 19 52 16

# After

Code 2024-09-29 19 51 16

Code 2024-09-29 19 52 39

**Review by rarkins — APPROVED** [review]

File: test/http-mock.ts [hunk 1]
```diff
@@ -77,30 +106,147 @@ export function scope(basePath: BasePath, options?: nock.Options): nock.Scope {
...
-  if (!nock.isDone()) {
-    errorLines.push('Unused mocks:');
-    errorLines.push(...nock.pendingMocks().map((x) => `  ${x}`));
+
+  const state = expect.getState();
+  if (!state.currentTestName || !state.testPath) {
+    return;
+  }
+
+  const content = fs.readFileSync(state.testPath, { encoding: 'utf8' });
```
Comment (viceice) [inline test/http-mock.ts:127]: this fails if fs-extra is mocked inside test

Comment (zharinov) [reply test/http-mock.ts:127]: Good catch

**Pull request merged**